### ISSUE TITLE
LinuxKit Kernel Build Dockerfile Error

### ISSUE THREAD

**Issue by ndauten** [linuxkit/linuxkit#2081]
**Description**

@justincormack trying to build custom kernel directory in LinuxKit. Getting some errors with default options. 

**Steps to reproduce the issue:**

1. Checkout latest linuxkit
2. cd kernel
3. make

**Describe the results you received:**

> docker pull linuxkit/kernel:4.11.5- || docker build --build-arg KERNEL_VERSION=4.11.5 --build-arg KERNEL_SERIES=4.11.x --build-arg DEBUG= --no-cache -t linuxkit/kernel:4.11.5- .
Error response from daemon: manifest for linuxkit/kernel:4.11.5- not found
Sending build context to Docker daemon 868.9 kB
Step 1/26 : FROM linuxkit/alpine:4f0ddee221c46f142e5a190dd43d2e07256ef98d AS kernel-build
Error parsing reference: "linuxkit/alpine:4f0ddee221c46f142e5a190dd43d2e07256ef98d AS kernel-build" is not a valid repository/tag: invalid reference format
Makefile:96: recipe for target 'build_4.11.x' failed
make: *** [build_4.11.x] Error 1

**Describe the results you expected:**

I expected the Docker engine to pull the linuxkit/alpine version down. 

**Additional information you deem important (e.g. issue happens only occasionally):**

In general I'm trying to build a custom kernel. It seems like the goal is to use a base alpine docker image, then inside of it download and build the custom kernel image. Is this true? 

Thanks!

**Comment by riyazdf** [comment]
@ndauten: from the docker pull command it looks like the git hash command isn't working properly in your Makefile.  The hash resolves and appends to the tag so that it's of the form: `linuxkit/kernel:4.11.5-7310f2ed0123b0cfe97c253300641f72662bee36` 

What platform are you building this on, and which version of `make` are you using? (`make -v`)

Also, which version of Docker are you using?  We use multistage builds for most of our packages, which is available in Docker 17.05 and above

**Comment by ndauten** [comment]
Linux hyperstache 3.16.0-4-amd64 #1 SMP Debian 3.16.39-1+deb8u2 (2017-03-07) x86_64 GNU/Linux

GNU Make 4.0

Docker version 17.03.1-ce, build c6d412e

So the Docker version is the problem then?

**Comment by ndauten** [comment]
But @riyazdf shouldn't it still be able to manually build from the Dockerfile? 

So it gives the error: 

```
Error parsing reference: "linuxkit/alpine:4f0ddee221c46f142e5a190dd43d2e07256ef98d AS kernel-build"
'''

Shouldn't that work? Or is that the issue with the multistage builds and Docker version?

**Comment by riyazdf** [comment]
@ndauten: the `linuxkit/alpine` build failure is indeed because your Docker install isn't compatible with multistage builds - upgrading Docker should fix that. 

I'm not sure what's going on with the hash/make failure.  Do you see any failures in the build logs that could have come from this assignment?  It seems that this is returning empty:
`HASH?=$(shell git ls-tree HEAD -- ../$(notdir $(CURDIR)) | awk '{print $$3}')`

**Comment by ijc** [comment]
> I'm not sure what's going on with the hash/make failure. Do you see any failures in the build logs that could have come from this assignment? It seems that this is returning empty:
> `HASH?=$(shell git ls-tree HEAD -- ../$(notdir $(CURDIR)) | awk '{print $$3}')`

Interesting, that's the same symptom as the CI failure #2063 in which I haven't been able to reproduce or explain (and neither has @vdemeester or anyone else AFAIK)

**Comment by ndauten** [comment]
@ijc I will try and take a look at tracing the environment to help debug the HASH. 

Is there any information or tests you can think of that would be of value?

**Comment by ijc** [comment]
TL;DR: Your git is too old.

See https://github.com/linuxkit/linuxkit/pull/2063#issuecomment-310382498 and https://github.com/linuxkit/linuxkit/pull/2063#issuecomment-310383498

**Comment by ndauten** [comment]
Ah excellent find! Is there a planned fix, or do we just have to upgrade git?

**Comment by ijc** [comment]
It's not clear yet.

**Comment by ndauten** [comment]
@ijc do you think we should close this issue now? I updated got and docker. Things work now.

**Comment by rn** [comment]
@ndauten thanks for the feedback. I'm closing this. The docker version issue is known, but will go away, not that 17.06 is out. Git version may be fixed by #2117
